### PR TITLE
Add `mssql` integration tests and relocate existing unit tests

### LINKED ISSUE #38577: Add `mssql` test integration
### Body

After removal Microsoft SQL Server as DB Backend (https://lists.apache.org/thread/pgcgmhf6560k8jbsmz8nlyoxosvltph2) we lost an ability to run unit tests over MS SQL database. All unit tests which rely on mssql moved into the integration tests (see https://github.com/apache/airflow/pull/38530):
- `tests/providers/apache/hive/transfers/test_mssql_to_hive.py` -> [`tests/integration/providers/apache/hive/transfers/test_mssql_to_hive.py`](https://github.com/apache/airflow/blob/main/tests/integration/providers/apache/hive/transfers/test_mssql_to_hive.py)
- `tests/providers/google/cloud/transfers/test_bigquery_to_mssql.py` -> [`tests/integration/providers/google/cloud/transfers/test_bigquery_to_mssql.py`](https://github.com/apache/airflow/blob/main/tests/integration/providers/google/cloud/transfers/test_bigquery_to_mssql.py)
- `tests/providers/google/cloud/transfers/test_mssql_to_gcs.py` -> [`tests/integration/providers/google/cloud/transfers/test_mssql_to_gcs.py`](https://github.com/apache/airflow/blob/main/tests/integration/providers/google/cloud/transfers/test_mssql_to_gcs.py)

We should eventually add MS SQL as available option for integration tests, and make to possible to run integration tests with it

### Committer

- [X] I acknowledge that I am a maintainer/committer of the Apache Airflow project.

### PR DESCRIPTION
**Pull request by shahar1**
closes: #38577

This PR:
- Relocates existing unit tests for `mssql` from the integration directory back to the unit tests (see #39768)
- Creates a Docker Compose integration yaml file for `mssql`.
- Adds an integration test for `mssql` hook.
- Adds integration tests for:
    - `bigquery_to_mssql.py`
    - `mssql_to_gcs.py`
    - `mssql_to_hive.py`

cc: @potiuk, @Taragolis, @eladkal

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: scripts/ci/docker-compose/integration-mssql.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+---
+version: "3.8"
+services:
+  mssql:
+    container_name: mssql
+    image: "mcr.microsoft.com/mssql/server:2017-latest"
```
Comment (shahar1) [inline scripts/ci/docker-compose/integration-mssql.yml]: I chose this version arbitrarily - should I replace it with the latest version / create a container for each `mssql` version and test for all versions in a matrix? (might be an overkill, but it's possible)

Comment (potiuk) [reply scripts/ci/docker-compose/integration-mssql.yml]: Overkill. We should possibly just bump it to some later version (2017 is really old) - likely 2019 should be good and supported for few more years. We have no promises on what version of mssql we support.

File: tests/integration/providers/google/cloud/transfers/test_mssql_to_gcs.py [hunk 2]
```diff
@@ -17,102 +17,65 @@
...
-            (-2, -2),
-            (datetime.date(1970, 1, 2), "1970-01-02"),
-            (datetime.date(1000, 1, 2), "1000-01-02"),
-            (datetime.datetime(1970, 1, 1, 1, 0), "1970-01-01T01:00:00"),
-            (datetime.time(hour=0, minute=0, second=0), "00:00:00"),
-            (datetime.time(hour=23, minute=59, second=59), "23:59:59"),
-        ],
-    )
-    def test_convert_type(self, value, expected):
+    def setup_method(self):
```
Comment (potiuk) [inline tests/integration/providers/google/cloud/transfers/test_mssql_to_gcs.py:45]: ~I think those tests should be moved to "mssql" provider. There is no particular reason they should be in "google" provider if they are change into integration tests.~

As @eladkal mentioned - it's somethign of a past that I remembered wrongly. It's good as-is right now.

**Comment by potiuk** [comment]
Nice. - few nits and solving failing tests (I guess some rebase needed?)

File: tests/integration/providers/google/cloud/transfers/test_mssql_to_gcs.py (hunk 2 above)
Comment (eladkal) [reply tests/integration/providers/google/cloud/transfers/test_mssql_to_gcs.py:45]: isn't the convention to have the test file in similar path to the code file?

Comment (potiuk) [reply tests/integration/providers/google/cloud/transfers/test_mssql_to_gcs.py:45]: Yes. You are right. I remembered it wrongly from the past.

I think in the past we had the convention that integration tests were related to particular integration were in subfolder related to particular integration, but those times are gone and indeed currently the structure is more similar to unit tests and our `integration` tests are now structured similarly to unit tests.

Comment (shahar1) [reply tests/integration/providers/google/cloud/transfers/test_mssql_to_gcs.py:45]: Cool, I'll keep it as it then

**Comment by potiuk** [comment]
One more thing: you also need to add Mssql as individual integration to test in `.github/workflows/integration-tests.yml`. This way the integration tests will be run on public runners (only in self-hosted runners all integrations are run together).

File: .github/workflows/integration-tests.yml [hunk 3]
```diff
@@ -140,6 +140,11 @@ jobs:
           breeze testing integration-tests --integration qdrant
           breeze down
         if: inputs.is-airflow-runner != 'true' && matrix.backend == 'postgres'
+      - name: "Integration Tests Postgres: mssql"
+        run: |
+          breeze testing integration-tests --integration mssql
+          breeze down
+        if: inputs.is-airflow-runner != 'true' && matrix.backend == 'mssql'
```
Comment (potiuk) [inline .github/workflows/integration-tests.yml]:
```suggestion
        if: inputs.is-airflow-runner != 'true' && matrix.backend == 'postgres'
```

**Comment by potiuk** [comment]
Nice!

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
@eladkal @shahar1 I also added #39866 as follow-up -> I realized that the last step was not obvious (having to add integration test step to integration-tests.yml was quite a bit duplication - also recently the "self-hosted" "all-testable" integrations when all of the integrations are started together have been somewhat flaky (as opposed to individual integrations) - so the #39866 will simplify it: removes the duplication and always runs the integration tests individually on public runners for both mssql and postgres.

**Pull request merged**